src/script/scriptBuilder.zig

```diff
@@ -0,0 +1,96 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const Script = @import("lib.zig").Script;
+const StackError = @import("stack.zig").StackError;
+pub const engine = @import("engine.zig");
+const Opcode = @import("./opcodes/constant.zig").Opcode;
+const isUnnamedPushNDataOpcode = @import("opcodes/constant.zig").isUnnamedPushNDataOpcode;
+const ScriptNum = @import("lib.zig").ScriptNum;
+const asBool = Script.asBool;
+const asInt = Script.asInt;
+const testing = std.testing;
+
+/// Maximum script length in bytes
+const MAX_SCRIPT_SIZE = 10000;
+
+///ScriptBuilder is a library to generate easier scripts, useful for faster testing
+pub const ScriptBuilder = struct {
+    /// Number of opcodes in the script(this is not a necessary field and may be removed later)
+    size: ?usize,
+
+    /// Dynamic array holding the opcodes
+    script: std.ArrayList(u8),
+
+    ///Memory Allocator
+    allocator: Allocator,
+
+    /// Initialize a new ScriptBuilder
+    ///
+    /// # Arguments
+    /// - `allocator`: Memory allocator for managing engine resources
+    ///
+    /// # Returns
+    /// - `!ScriptBuilder`: use with `try`
+    pub fn new(allocator: Allocator, opcodeCount: ?u8) !ScriptBuilder {
+        return ScriptBuilder{
+            .size = opcodeCount orelse null,
+            .script = std.ArrayList(u8).init(allocator),
+            .allocator = allocator,
+        };
+    }
+
+    ///Push an OPcode to the ScriptBuilder
+    pub fn addOpcode(self: *ScriptBuilder, op: Opcode) !*ScriptBuilder {
+        if (self.script.items.len + 1 > MAX_SCRIPT_SIZE) {
+            return StackError.OutOfMemory;
```

Comment:
It is not a stack error. The script and the stack are two different things. During the script execution, the stack is used.

Create a const ScriptBuilderError = error {ScriptTooLong};